File: src/main/java/com/team04/back/domain/weather/weather/dto/WeatherInfoDto.java
package com.team04.back.domain.weather.weather.dto;

import com.team04.back.domain.weather.weather.entity.WeatherInfo;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;


public record WeatherInfoDto(
        @NotNull int id,
        @NotNull String weather,
        Integer weatherCode,
        String weatherDescription,
        @NotNull Double dailyTemperatureGap,
        @NotNull Double feelsLikeTemperature,
        @NotNull Double maxTemperature,
        @NotNull Double minTemperature,
        Double pop,
        Double rain,
        Double snow,
        Integer humidity,
        Double windSpeed,
        Integer windDeg,
        Double uvi,
        @NotNull String location,
        @NotNull LocalDate date
) {
    public WeatherInfoDto(WeatherInfo weatherInfo) {
        this (
                weatherInfo.getId(),
                weatherInfo.getWeather().name(),
                weatherInfo.getWeather().getCode(),
                weatherInfo.getDescription(),
                weatherInfo.getDailyTemperatureGap(),
                weatherInfo.getFeelsLikeTemperature(),
                weatherInfo.getMaxTemperature(),
                weatherInfo.getMinTemperature(),
                weatherInfo.getPop(),
                weatherInfo.getRain(),
                weatherInfo.getSnow(),
                weatherInfo.getHumidity(),
                weatherInfo.getWindSpeed(),
                weatherInfo.getWindDeg(),
                weatherInfo.getUvi(),
                weatherInfo.getLocation(),
                weatherInfo.getDate()
        );
    }
}

File: src/main/java/com/team04/back/domain/weather/weather/dto/WeatherInfoDto.kt
package com.team04.back.domain.weather.weather.dto

import com.team04.back.domain.weather.weather.entity.WeatherInfo
import java.time.LocalDate

data class WeatherInfoDto(
    val id: Int,
    val weather: String,
    val weatherCode: Int,
    val weatherDescription: String?,
    val dailyTemperatureGap: Double,
    val feelsLikeTemperature: Double,
    val maxTemperature: Double,
    val minTemperature: Double,
    val pop: Double?,
    val rain: Double?,
    val snow: Double?,
    val humidity: Int?,
    val windSpeed: Double?,
    val windDeg: Int?,
    val uvi: Double?,
    val location: String,
    val date: LocalDate
) {
    constructor(weatherInfo: WeatherInfo) : this(
        weatherInfo.id,
        weatherInfo.weather.name,
        weatherInfo.weather.code,
        weatherInfo.description,
        weatherInfo.dailyTemperatureGap,
        weatherInfo.feelsLikeTemperature,
        weatherInfo.maxTemperature,
        weatherInfo.minTemperature,
        weatherInfo.pop,
        weatherInfo.rain,
        weatherInfo.snow,
        weatherInfo.humidity,
        weatherInfo.windSpeed,
        weatherInfo.windDeg,
        weatherInfo.uvi,
        weatherInfo.location,
        weatherInfo.date
    )
}